src/app/shared/components/template/components/task-progress-bar/task-progress-bar.component.ts

```diff
@@ -1,81 +1,175 @@
-import { Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from "@angular/core";
 import { TaskService } from "src/app/shared/services/task/task.service";
-import { getStringParamFromTemplateRow } from "src/app/shared/utils";
+import {
+  getBooleanParamFromTemplateRow,
+  getStringParamFromTemplateRow,
+} from "src/app/shared/utils";
 import { TemplateBaseComponent } from "../base";
 import { TemplateFieldService } from "../../services/template-field.service";
 import { AppDataService } from "src/app/shared/services/data/app-data.service";
 import { IProgressStatus } from "src/app/shared/services/task/task.service";
 import { CampaignService } from "src/app/feature/campaign/campaign.service";
+import { Subscription, debounceTime } from "rxjs";
+import { DynamicDataService } from "src/app/shared/services/dynamic-data/dynamic-data.service";
+
+interface ITaskProgressBarParams {
+  /**
+   * TEMPLATE PARAMETER: task_group_data.
+   * The name of the task group to track the progress of (through the completion of its subtasks)
+   */
+  taskGroupDataList: string;
+  /**
+   * TEMPLATE PARAMETER: completed_field.
+   * The name of the field that stores the completion status of the task group
+   */
+  taskGroupCompletedField: string;
+  /**
+   * TEMPLATE PARAMETER: progress_units_name.
+   * The displayed name of the units of progress associated with
+   * the constituent tasks of the task group.
+   * Default "sections"
+   */
+  progressUnitsName: string;
+  /**
+   * TEMPLATE PARAMETER: show_text.
+   * Show the text associated with progress bar, e.g. "8 sections"
+   * Default false
+   */
+  showText: boolean;
+  /**
+   * TEMPLATE PARAMETER: variant.
+   * A list of named style variants of the component, separated by spaces or commas.
+   * */
+  variant: "green" | "secondary";
```

Comment:
nit(blocking) - mentioned in the scss but now looking at the options laid out really doesn't quite feel right as a list of options - so would say more blocking.

Reply:
@jfmcquade - I can't see any new commits, could you clarify how this has been resolved?

Reply:
Sorry, I was working through the comments and marking them as resolved as I made changes locally, but I haven't pushed my changes yet as I thought it would be better to wait until I'm ready to request another review. I thought better to hold off rather than give you lots of notifications before I've finished addressing all the point – but I'm guessing that you get notifications for me marking comments as resolved anyway? In future, I'll keep track of the comments locally and only mark as resolved on github when I've pushed the corresponding changes